torch/utils/data/dataset.py

```diff
@@ -73,6 +73,8 @@ def __len__(self):
         return self.cumulative_sizes[-1]
 
     def __getitem__(self, idx):
+        if idx < 0:
+            idx = self.__len__() + idx
```

Comment:
Please use `len(self)`. Also, it would be nice to add a check for `-idx <= len(self)` and raise an error otherwise

Reply:
Hi Adam - sorry I missed this. Added another commit.